tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,167 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import netaddr
+import yaml
+from nose_parameterized import parameterized
+
+from tests.st.test_base import TestBase
+from tests.st.utils.docker_host import DockerHost
+from tests.st.utils.exceptions import CommandExecError
+from tests.st.utils.utils import ETCD_CA, ETCD_CERT, \
+    ETCD_KEY, ETCD_HOSTNAME_SSL, ETCD_SCHEME, get_ip, wipe_etcd, retry_until_success
+
+if ETCD_SCHEME == "https":
+    ADDITIONAL_DOCKER_OPTIONS = "--cluster-store=etcd://%s:2379 " \
+                                "--cluster-store-opt kv.cacertfile=%s " \
+                                "--cluster-store-opt kv.certfile=%s " \
+                                "--cluster-store-opt kv.keyfile=%s " % \
+                                (ETCD_HOSTNAME_SSL, ETCD_CA, ETCD_CERT,
+                                 ETCD_KEY)
+else:
+    ADDITIONAL_DOCKER_OPTIONS = "--cluster-store=etcd://%s:2379 " % \
+                                get_ip()
+
+
+class TestDefaultPools(TestBase):
+
+    @classmethod
+    def setUpClass(cls):
+        # First, create a host to run things in
+        cls.hosts = []
+        cls.host = DockerHost("host",
+                              additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
+                              start_calico=False)
+        cls.hosts.append(cls.host)
+
+    def setUp(self):
+        try:
+            self.host.execute("docker rm -f calico-node")
+        except CommandExecError:
+            # Presumably calico-node wasn't running
+            pass
+        wipe_etcd(get_ip())
+
+    @parameterized.expand([
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/27",  None),    # Too small
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/32",  None),    # Too small, but legal
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/33",  None),    # Impossible CIDR
+        (False, "CALICO_IPV4POOL_CIDR", "256.0.0.0/24", None),    # Invalid IP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24",  None),    # Typical non-default pool
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/26",  None),    # Smallest legal pool
+        (True,  "CALICO_IPV6POOL_CIDR", "fd00::/122",   None),    # Smallest legal pool
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123",   None),    # Too small
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128",   None),    # Too small, but legal
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129",   None),    # Impossible CIDR
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "cross-subnet"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "always"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "off"),  # Typical pool, explicitly no IPIP
```

Comment:
Could have another test case with an inalid string for the IPIP mode?  Not too necessary to add because I know there is one at the unit test level but just wanted to mention for completeness.